### DISCUSSION TITLE
Large text files

### DISCUSSION THREAD

**Discussion by maxbaluev** [supabase/supabase#2517] [Questions]
We are building book text editor with lazy loading, is superbase suitable to store and modify large text files on the fly ?

**Comment by inian** [comment]
Hi @maxbaluev, storing large text files is better suited in [Supabase Storage](https://supabase.io/storage) instead of directly storing them in your database, but also depends on how large your text files are

**Reply by maxbaluev** [reply]
Hello, ty for reply 👍 Our text files are all kinds of books. But we need not only to store data, but modify too. For example user can edit one word on the middle of the book on the frontend editor and we need to save changes. In addition, this process can be repeated a large number of times in a short time.

**Reply by inian** [reply]
When the user edits the book, are you sending the entire modified book text again to the server? If you are, using the [storage upsert](https://supabase.io/docs/reference/javascript/storage-from-upload) function would be a good fit and storage would scale better for larger books. 

If you are somehow sending only the delta of the books and have an efficient datastructure to store the modified books and the size of the books are small, then storing them in the database might work. 

Also note that the [pricing](https://supabase.io/pricing) for storage is cheaper than pricing for database space
